tools/linux_vsts.ps1

```diff
@@ -0,0 +1,300 @@
+        # If the Build number has a dot in it, only the string before dot is inserted into releaseTag.
+        $buildMajorNumber = $env:BUILD_BUILDNUMBER
+        $dotPos = $buildMajorNumber.IndexOf(".")
+        if ($dotPos -ne -1) {
+            $buildMajorNumber = $buildMajorNumber.Substring(0, $dotPos)
+        }
+        $releaseTag = $releaseTag+'.'+$buildMajorNumber
+    }
+
+    return $releaseTag
+}
+
+# This function retrieves the appropriate svg to be used when presenting
+# the daily test run badge
+# the location in azure is public readonly
+function Get-DailyBadge
+{
+    param (
+        [Parameter(Mandatory=$true,Position=0)][ValidateSet("Pass","Fail")]$result
+    )
+    $PASS = "https://jimtru1979.blob.core.windows.net/badges/DailyBuild.Pass.svg"
+    $FAIL = "https://jimtru1979.blob.core.windows.net/badges/DailyBuild.Fail.svg"
+
+    if ( $result -eq "Pass" ) { $BadgeUrl = $PASS } else { $BadgeUrl = $FAIL }
+    $response = Invoke-WebRequest -Uri $BadgeUrl
+    if ( $response.StatusCode -ne 200 ) { throw "Could not read badge '$BadgeUrl'" }
+    $response.Content
+}
+
+# This function uses Azure REST api to update the daily test pass results
+# it relies on writing a specific SVG into a constant location so the
+# README.MD can report on the status of the daily test pass
+# it also relies on two environment variables which need to be set in the
+# Travis-CI config which is the account name and key for the azure blob location
+#
+# the best way to do this would be if travis-ci supported a webcall to get
+# the status of cron_job builds, but it doesn't, so we have this
+# also, since we can have a build on Linux which succeeds and one on macOS which
+# doesn't we'll set the appropriate badge so the the README can pick it up
+function Set-DailyBuildBadge
+{
+    [CmdletBinding(SupportsShouldProcess=$true)]
+    param ( [Parameter(Mandatory=$true,Position=0)]$content )
+    $method = "PUT"
+    $headerDate = '2015-12-11'
+
+    $storageAccountName = $Env:TestResultAccountName
+    $storageAccountKey = $Env:TestResultAccountKey
+
+    # this is the url referenced in README.MD which displays the badge
+    $platform = "Linux"
+    $Url = "https://jimtru1979.blob.core.windows.net/badges/DailyBuildStatus.${platform}.svg"
+
+    $body = $content
+    $bytes = ([System.Text.Encoding]::UTF8.GetBytes($body))
+    $contentLength = $bytes.length
+
+    $now = [datetime]::UtcNow.ToString("R", [System.Globalization.CultureInfo]::InvariantCulture)
+    $headers = @{
+        "x-ms-date"      = $now
+        "cache-control"  = "no-cache"
+        "x-ms-blob-type" = "BlockBlob"
+        "x-ms-version"   = "$headerDate"
+    }
+
+    $contentType = "image/svg+xml"
+    # more info: https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/put-blob
+    $sb = [text.stringbuilder]::new()
+    # can't use AppendLine because the `r`n causes the command to fail, it must be `n and only `n
+    $null = $sb.Append("$method`n")
```

Comment:
` [System.Environment]::NewLine` might be better than `n. But if stringbuilder uses `r`n on Linux, this sounds more like a bug in .Net Core (or maybe I misunderstood its usage)